### PR TITLE
Export libq public headers

### PR DESCRIPTION
**Pull request by jupp0r**
In order to more easily consume libq from other CMake projects,
this change exposes the public headers as part of the library target.
This automatically sets target_include_directories for targets that
link against libq. See https://rix0r.nl/blog/2015/08/13/cmake-guide/
for more details.

### CODE REVIEW

File: libs/q/CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,20 @@
-
 find_source_tree( LIBQ_HEADERS          "Header Files"          include/q "*.hpp" )
 find_source_tree( LIBQ_INTERNAL_HEADERS "Internal Header Files" src       "*.hpp" )
 find_source_tree( LIBQ_SOURCES          "Source Files"          src       "*.cpp" )
 
-add_library( q ${LIBQ_HEADERS} ${LIBQ_INTERNAL_HEADERS} ${LIBQ_SOURCES} )
+add_library( q ${LIBQ_SOURCES} )
```
Comment (grantila) [inline libs/q/CMakeLists.txt]: This removes the header files from the project, when working in an IDE (Visual Studio, XCode, etc).
I see in the article you linked to that this is the way it "should be done". There's _certainly_ something wrong with this approach.

Comment (jupp0r) [inline libs/q/CMakeLists.txt]: You are right, I added `source_group` statements to remedy this problem.

**Comment by grantila** [comment]
What are you doing in _your_ cmake project to include q? I'm gonna spend some time making sure this works fine also for `q-test` as well as when packaging q. I haven't thought much about using q as a cmake-dependency as you've realized, but it's a good point to be able to.

**Comment by jupp0r** [comment]
I'm writing a library that returns promises on its public interface.  The exported config for `libq` allow consumers of my library to automatically include `libq` headers, linker flags, etc. All I'd have to do is

```
add_subdirectory(external/q)
target_link_libraries(my_library q)
```

When you have a deep dependency tree, it's increasingly important to let CMake handle `include_directories` and linker flags for you.

Regarding `q-test`, would you be interested in a patch that applies the same technique there?

**Comment by grantila** [comment]
@jupp0r yes that would be needed (or at least very much preferred) to have both `q` and `q-test` exported. I've tried, without success, since the google test's `gtest_main` is not in the "export set". So I get this:
```
CMake Error: install(EXPORT "QTestConfig" ...) includes target "q-test" which requires target "gtest_main" that is not in the export set.
```

Your help is highly appreciated!

**Comment by grantila** [comment]
Note, this is in the `cmake-targets` branch

**Pull request merged**